src/bugpoint.rs

```diff
@@ -0,0 +1,312 @@
+//! CLI tool to read Cranelift IR files and compile them into native code.
+
+use crate::disasm::{PrintRelocs, PrintTraps};
+use crate::utils::{parse_sets_and_triple, read_to_string};
+use cranelift_codegen::ir::{Ebb, Function, Inst, InstBuilder, TrapCode};
+use cranelift_codegen::isa::TargetIsa;
+use cranelift_codegen::settings::FlagsOrIsa;
+use cranelift_codegen::timing;
+use cranelift_codegen::Context;
+use cranelift_reader::parse_test;
+use std::path::Path;
+use std::path::PathBuf;
+
+use indicatif::{ProgressBar, ProgressDrawTarget, ProgressStyle};
+
+pub fn run(
+    filename: &str,
+    flag_set: &[String],
+    flag_isa: &str,
+    verbose: bool,
+) -> Result<(), String> {
+    let parsed = parse_sets_and_triple(flag_set, flag_isa)?;
+    let fisa = parsed.as_fisa();
+
+    let path = Path::new(&filename).to_path_buf();
+    let name = String::from(path.as_os_str().to_string_lossy());
```

Comment:
Why is the round trip from filename to name needed, here?

Reply:
It isnt I think. I copy pasted and inlined some functions from elsewhere.